instrumentation/spring-webclient-5.0/src/main/java/com/nr/agent/instrumentation/spring_webclient/Util.java

```diff
@@ -22,35 +26,49 @@
 public class Util {
 
     private static final String LIBRARY = "Spring-WebClient";
+    private static final String SEGMENT_ATTRIBUTE = "newrelic-segment";
     private static final URI UNKNOWN_HOST = URI.create("UnknownHost");
-    private static URI uri = null;
 
-    public static BiConsumer<? super ClientResponse, Throwable> reportAsExternal(Segment segment) {
+    public static void startExternalSegmentIfNeeded(WebClient.RequestHeadersSpec<?> request) {
+        Transaction txn = AgentBridge.getAgent().getTransaction(false);
+        if (txn != null) {
+            Segment segment = txn.startSegment("WebClient.exchange");
+            segment.addOutboundRequestHeaders(new OutboundRequestWrapper(request));
+            request.attribute(SEGMENT_ATTRIBUTE, segment);
+        }
+    }
+
+    public static Mono<ClientResponse> reportAsExternal(ClientRequest request, Mono<ClientResponse> response) {
+        Segment segment = (Segment) request.attribute(SEGMENT_ATTRIBUTE).orElse(null);
```

Comment:
I think you may need to do a null check on request, since this will run before the original code

Reply:
@tspring done